### DISCUSSION TITLE
From bit to np.ndarray

### DISCUSSION THREAD

**Discussion by cpsaroud** [QuSpin/QuSpin#428] [Q&A] [answered]
It might be a trivial question, but is there a simple way to calculate the overlap of a Hamiltonian's eigenstates with the basis states used to construct this Hamiltonian? (basis states are in a bit representation, while eigenstates are numpy.ndarray

**Comment by JiahaoYao** [comment, marked as answer]
Hi @cpsaroud, thanks for your questions and your interest in quspin! yes and you might find interest in [index method](https://weinbe58.github.io/QuSpin/generated/quspin.basis.boson_basis_1d.html?highlight=index#quspin.basis.boson_basis_1d.index). 

For example, you can call `basis.index(bit_string)` to get the index: 

```python
i0 = index("111000") # pick state from basis set
print(basis)
print(i0)
psi = np.zeros(basis.Ns,dtype=np.float64)
psi[i0] = 1.0 # define state corresponding to the string "111000"
```

**Reply by cpsaroud** [reply]
Yes, it works. Thanks a lot
